core/lib/server/pub.ts

```diff
@@ -1398,6 +1480,113 @@ export async function getPubsWithRelatedValuesAndChildren<
 						)
 					)
 			)
+			.with("stage_ms", (db) =>
+				db
+					.selectFrom("stage_memberships")
+					.$if(Boolean(props.userId), (qb) =>
+						qb
+							.where("stage_memberships.userId", "=", props.userId!)
+							.select([
+								"role",
+								"stageId as membId",
+								sql<MembershipType>`'stage'::"MembershipType"`.as("type"),
+							])
+					)
+					.$castTo<{
+						role: MembershipCapabilitiesRole;
+						membId: string;
+						type: MembershipType;
+					}>()
+			)
+			.with("pub_ms", (db) =>
+				db
+					.selectFrom("pub_memberships")
+					.$if(Boolean(props.userId), (qb) =>
+						qb
+							.where("pub_memberships.userId", "=", props.userId!)
+							.select([
+								"role",
+								"pubId as membId",
+								sql<MembershipType>`'pub'::"MembershipType"`.as("type"),
+							])
+					)
+					.$castTo<{
+						role: MembershipCapabilitiesRole;
+						membId: string;
+						type: MembershipType;
+					}>()
+			)
+			.with("community_ms", (db) =>
+				db
+					.selectFrom("community_memberships")
+					.$if(Boolean(props.userId), (qb) =>
+						qb
+							.where("community_memberships.userId", "=", props.userId!)
+							.where("community_memberships.communityId", "=", props.communityId)
+							.select([
+								"role",
+								"communityId as membId",
+								sql<MembershipType>`'community'::"MembershipType"`.as("type"),
+							])
+					)
+					// Add a fake community admin role when selecting without a userId
+					.$if(!Boolean(props.userId), (qb) =>
+						qb.select((eb) => [
+							sql<MemberRole>`${MemberRole.admin}::"MemberRole"`.as("role"),
+							eb.val(props.communityId).as("membId"),
+							sql<MembershipType>`'community'::"MembershipType"`.as("type"),
+						])
+					)
+
+					.$castTo<{
+						role: MembershipCapabilitiesRole;
+						membId: string;
+						type: MembershipType;
+					}>()
```

Comment:
These three CTEs all use this cast so they can return rows with the same schema which allows us to combine the rows with a `union` in the next CTE.

Reply:
So `membId` can be a stage id, pub id, or community id, and `type` tells us which